// @tgsnake/core/lib/src/errors/File.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileIsNotReadable = exports.FileUploadBigger = exports.FileUploadZero = exports.FileError = void 0;
const Base_js_1 = require("./Base.js");
class FileError extends Base_js_1.BaseError {
    constructor(message, description) {
        super();
        this.message = message;
        this.description = description;
    }
}
exports.FileError = FileError;
class FileUploadZero extends FileError {
    constructor() {
        super("Can't upload file when it zero bytes.", 'Provided file has zero bytes (0 B) file size.');
    }
}
exports.FileUploadZero = FileUploadZero;
class FileUploadBigger extends FileError {
    constructor(limit, size) {
        super(`File greater than ${limit} B.`, `The provided file has ${size} B file size, it greater than ${limit} B`);
    }
}
exports.FileUploadBigger = FileUploadBigger;
class FileIsNotReadable extends FileError {
    constructor() {
        super('FILE_IS_NOT_READABLE', 'The argument provided is not a Readable stream.');
    }
}
exports.FileIsNotReadable = FileIsNotReadable;
