// @tgsnake/core/lib/src/errors/exceptions/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceUnavailable = exports.InternalServerError = exports.Flood = exports.NotAcceptable = exports.NotFound = exports.Forbidden = exports.Unauthorized = exports.BadRequest = exports.SeeOther = exports.Exceptions = exports.count = void 0;
var All_js_1 = require("./All.js");
Object.defineProperty(exports, "count", { enumerable: true, get: function () { return All_js_1.count; } });
Object.defineProperty(exports, "Exceptions", { enumerable: true, get: function () { return All_js_1.Exceptions; } });
exports.SeeOther = __importStar(require("./SeeOther303.js"));
exports.BadRequest = __importStar(require("./BadRequest400.js"));
exports.Unauthorized = __importStar(require("./Unauthorized401.js"));
exports.Forbidden = __importStar(require("./Forbidden403.js"));
exports.NotFound = __importStar(require("./NotFound404.js"));
exports.NotAcceptable = __importStar(require("./NotAcceptable406.js"));
exports.Flood = __importStar(require("./Flood420.js"));
exports.InternalServerError = __importStar(require("./InternalServerError500.js"));
exports.ServiceUnavailable = __importStar(require("./ServiceUnavailable503.js"));
