// @tgsnake/core/lib/src/errors/exceptions/SeeOther303.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserMigrate = exports.StatsMigrate = exports.PhoneMigrate = exports.NetworkMigrate = exports.FileMigrate = exports.SeeOther = void 0;
const RpcError_js_1 = require("../RpcError.js");
class SeeOther extends RpcError_js_1.RPCError {
    code = 303;
    name = 'SEE_OTHER';
}
exports.SeeOther = SeeOther;
class FileMigrate extends SeeOther {
    id = 'FILE_MIGRATE_X';
    message = 'The file to be accessed is currently stored in DC{value}';
}
exports.FileMigrate = FileMigrate;
class NetworkMigrate extends SeeOther {
    id = 'NETWORK_MIGRATE_X';
    message = 'Your IP address is associated to DC {value}, please re-send the query to that DC.';
}
exports.NetworkMigrate = NetworkMigrate;
class PhoneMigrate extends SeeOther {
    id = 'PHONE_MIGRATE_X';
    message = 'Your phone number is associated to DC {value}, please re-send the query to that DC.';
}
exports.PhoneMigrate = PhoneMigrate;
class StatsMigrate extends SeeOther {
    id = 'STATS_MIGRATE_X';
    message = 'Channel statistics for the specified channel are stored on DC {value}, please re-send the query to that DC.';
}
exports.StatsMigrate = StatsMigrate;
class UserMigrate extends SeeOther {
    id = 'USER_MIGRATE_X';
    message = 'Your account is associated to DC {value}, please re-send the query to that DC.';
}
exports.UserMigrate = UserMigrate;
