// @tgsnake/core/lib/src/raw/core/primitive/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Float = exports.Double = exports.Vector = exports.BoolFalse = exports.BoolTrue = exports.Bool = exports.String = exports.Bytes = exports.Int256 = exports.Int128 = exports.Long = exports.Int = void 0;
var Int_js_1 = require("./Int.js");
Object.defineProperty(exports, "Int", { enumerable: true, get: function () { return Int_js_1.Int; } });
Object.defineProperty(exports, "Long", { enumerable: true, get: function () { return Int_js_1.Long; } });
Object.defineProperty(exports, "Int128", { enumerable: true, get: function () { return Int_js_1.Int128; } });
Object.defineProperty(exports, "Int256", { enumerable: true, get: function () { return Int_js_1.Int256; } });
var Bytes_js_1 = require("./Bytes.js");
Object.defineProperty(exports, "Bytes", { enumerable: true, get: function () { return Bytes_js_1.Bytes; } });
var String_js_1 = require("./String.js");
Object.defineProperty(exports, "String", { enumerable: true, get: function () { return String_js_1.String; } });
var Bool_js_1 = require("./Bool.js");
Object.defineProperty(exports, "Bool", { enumerable: true, get: function () { return Bool_js_1.Bool; } });
Object.defineProperty(exports, "BoolTrue", { enumerable: true, get: function () { return Bool_js_1.BoolTrue; } });
Object.defineProperty(exports, "BoolFalse", { enumerable: true, get: function () { return Bool_js_1.BoolFalse; } });
var Vector_js_1 = require("./Vector.js");
Object.defineProperty(exports, "Vector", { enumerable: true, get: function () { return Vector_js_1.Vector; } });
var Double_js_1 = require("./Double.js");
Object.defineProperty(exports, "Double", { enumerable: true, get: function () { return Double_js_1.Double; } });
var Float_js_1 = require("./Float.js");
Object.defineProperty(exports, "Float", { enumerable: true, get: function () { return Float_js_1.Float; } });
