// @tgsnake/core/lib/src/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Loggers = exports.Timeouts = exports.Helpers = exports.Versions = exports.Storages = exports.Sessions = exports.Raws = exports.Files = exports.Errors = exports.Cryptos = exports.Connections = exports.Clients = exports.TCP = exports.Queue = exports.Raw = exports.Client = void 0;
var Client_js_1 = require("./client/Client.js");
Object.defineProperty(exports, "Client", { enumerable: true, get: function () { return Client_js_1.Client; } });
var index_js_1 = require("./raw/index.js");
Object.defineProperty(exports, "Raw", { enumerable: true, get: function () { return index_js_1.Raw; } });
var Queue_js_1 = require("./Queue.js");
Object.defineProperty(exports, "Queue", { enumerable: true, get: function () { return Queue_js_1.Queue; } });
var connection_js_1 = require("./connection/connection.js");
Object.defineProperty(exports, "TCP", { enumerable: true, get: function () { return connection_js_1.TCP; } });
exports.Clients = __importStar(require("./client/index.js"));
exports.Connections = __importStar(require("./connection/index.js"));
exports.Cryptos = __importStar(require("./crypto/index.js"));
exports.Errors = __importStar(require("./errors/index.js"));
exports.Files = __importStar(require("./file/index.js"));
exports.Raws = __importStar(require("./raw/index.js"));
exports.Sessions = __importStar(require("./session/index.js"));
exports.Storages = __importStar(require("./storage/index.js"));
exports.Versions = __importStar(require("./Version.node.js"));
exports.Helpers = __importStar(require("./helpers.js"));
exports.Timeouts = __importStar(require("./Timeout.js"));
exports.Loggers = __importStar(require("./Logger.js"));
