// @tgsnake/core/lib/src/raw/core/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecretChatMessageService = exports.SecretChatMessage = exports.UpdateSecretChatMessage = exports.MsgContainer = exports.Message = exports.GzipPacked = exports.TLObject = exports.BytesIO = exports.Primitive = void 0;
const Primitive = __importStar(require("./primitive/index.js"));
exports.Primitive = Primitive;
var BytesIO_js_1 = require("./BytesIO.js");
Object.defineProperty(exports, "BytesIO", { enumerable: true, get: function () { return BytesIO_js_1.BytesIO; } });
var TLObject_js_1 = require("./TLObject.js");
Object.defineProperty(exports, "TLObject", { enumerable: true, get: function () { return TLObject_js_1.TLObject; } });
var GzipPacked_js_1 = require("./GzipPacked.js");
Object.defineProperty(exports, "GzipPacked", { enumerable: true, get: function () { return GzipPacked_js_1.GzipPacked; } });
var Message_js_1 = require("./Message.js");
Object.defineProperty(exports, "Message", { enumerable: true, get: function () { return Message_js_1.Message; } });
var MsgContainer_js_1 = require("./MsgContainer.js");
Object.defineProperty(exports, "MsgContainer", { enumerable: true, get: function () { return MsgContainer_js_1.MsgContainer; } });
var UpdateSecretChat_js_1 = require("./UpdateSecretChat.js");
Object.defineProperty(exports, "UpdateSecretChatMessage", { enumerable: true, get: function () { return UpdateSecretChat_js_1.UpdateSecretChatMessage; } });
Object.defineProperty(exports, "SecretChatMessage", { enumerable: true, get: function () { return UpdateSecretChat_js_1.SecretChatMessage; } });
Object.defineProperty(exports, "SecretChatMessageService", { enumerable: true, get: function () { return UpdateSecretChat_js_1.SecretChatMessageService; } });
