// @tgsnake/core/lib/src/raw/core/Message.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Message = void 0;
const BytesIO_js_1 = require("./BytesIO.js");
const TLObject_js_1 = require("./TLObject.js");
const Primitive = __importStar(require("./primitive/index.js"));
const platform_node_js_1 = require("../../platform.node.js");
function toBytes(value) {
    const bytesArray = [];
    for (let i = 0; i < 8; i++) {
        let shift = value >> BigInt(8 * i);
        shift &= BigInt(255);
        bytesArray[i] = Number(String(shift));
    }
    return platform_node_js_1.Buffer.from(bytesArray);
}
class Message extends TLObject_js_1.TLObject {
    static ID = 0x5bb8e511;
    msgId;
    seqNo;
    length;
    body;
    constructor(body, msgId, seqNo, length) {
        super();
        this.className = 'Message';
        this._slots = ['body', 'msgId', 'seqNo', 'length'];
        this.msgId = msgId;
        this.seqNo = seqNo;
        this.length = length;
        this.body = body;
    }
    static async read(data, ..._args) {
        const msgId = await Primitive.Long.read(data);
        const seqNo = await Primitive.Int.read(data);
        const length = await Primitive.Int.read(data);
        const body = data.read(length);
        return new Message(await TLObject_js_1.TLObject.read(new BytesIO_js_1.BytesIO(body)), msgId, seqNo, length);
    }
    write() {
        const bytes = new BytesIO_js_1.BytesIO();
        bytes.write(toBytes(this.msgId));
        bytes.write(Primitive.Int.write(this.seqNo));
        bytes.write(Primitive.Int.write(this.length));
        bytes.write(this.body.write());
        return platform_node_js_1.Buffer.from(bytes.buffer);
    }
}
exports.Message = Message;
